### ISSUE TITLE
Click methods with function pointer parameters on tooltip may make VS crash

### ISSUE THREAD

**Issue by Sunnie-Shine** [dotnet/roslyn#48233]
**Version Used**: 
VS 16.7.5

**Steps to Reproduce**:

1. Type code
```csharp
private static unsafe void M(delegate*<int, int> f) { ... }

private static int Call(int s) { ... }

private static unsafe int OtherOne()
{
    M(&Call);
}
```

2. Cursor move to above the method call `M(&Call)`, we can see the tooltip.
3. Click `M` in `M(&Call)` on the tooltip.

**Expected Behavior**:
Jump to the implementation.

**Actual Behavior**:
VS crashes.

**Comment by jasonmalinowski** [comment]
This works fine in current 16.8 Previews; @Sunnie-Shine please upgrade to 16.8, as you should see this and a bunch of other function pointer bugs fixed!

**Comment by Sunnie-Shine** [comment]
@jasonmalinowski I'm so sorry :( I don't know the bug had been fixed in 16.8 by the time I opened this issue! Thank you for your reply and suggestion!
